pulser/simulation/simulation.py

```diff
@@ -604,19 +702,21 @@ def build_coeffs_ops(basis: str, addr: str) -> list[list]:
             self.operators[addr][basis] = operators
             return terms
 
+        qobj_list = []
         # Time independent term:
-        if self.basis_name == "digital" or self._size == 1:
-            qobj_list = [0 * self._build_operator("I")]
-        else:
-            # Van der Waals Interaction Terms
-            qobj_list = [make_vdw_term()]
+        effective_size = self._size - sum(self._bad_atoms.values())
+        if self.basis_name != "digital" and effective_size > 1:
+            qobj_list.append(make_interaction_term())
 
         # Time dependent terms:
         for addr in self.samples:
             for basis in self.samples[addr]:
                 if self.samples[addr][basis]:
                     qobj_list += cast(list, build_coeffs_ops(basis, addr))
 
+        if not qobj_list:  # If qobj_list ends up empty
+            qobj_list = [0 * self.build_operator([("I", "global")])]
```

Comment:
I think this might be an unnecessary overhead, especially because it cycles over every qubit. Can something simpler like `qutip.Qobj(shape=(n, n))` be used instead?

Reply:
I agree, but I'm not sure that exists. Plus, bear in mind this will only be run for an empty sequence, which is the same as saying "almost never"